doc/basics.md

```diff
@@ -399,6 +399,36 @@ And another one:
 
 Notice how we can string several operation (`parser.parse(abstract_json)["str"]["123"]["abc"].get<double>()`) and only check for the error once, a strategy we call  *error chaining*.
 
+
+The next two functions will take as input a JSON document containing an array with a single element, either a string or a number. They return true upon success.
+
+```C++
+simdjson::dom::parser parser{};
+
+bool ParseDouble(const char *j, double &d) {
+    simdjson::error_code error;
+    parser.parse(j, std::strlen(j)).at(0).get<double>().tie(d, error);
+    if (error) {
+      return false;
+    }
+    return true;
+}
+
+bool ParseString(const char *j, std::string &s) {
+  simdjson::error_code error;
+  std::string_view answer;
+  parser.parse(j,strlen(j))
```

Comment:
This one could use `auto [answer, error] = ` if you so desire.

Also, it would probably be worth doing this multiline breakout of the parse() line in the other example too, just so people can eyeball them back and forth to see the differences quickly.

Reply:
Will do.

Reply:
Done. I did not reformat all of the prior examples. I just reformatted the new ones.